bitwarden_license/src/Commercial.Core/AdminConsole/Services/ProviderService.cs

```diff
@@ -387,10 +392,93 @@ public async Task AddOrganizationsToReseller(Guid providerId, IEnumerable<Guid>
             throw new BadRequestException("Organizations must not be assigned to any Provider.");
         }
 
+        await UpdatePlanForExistingOrgAsync(organizationIds, provider);
+
         var providerOrganizationsToInsert = organizationIds.Select(orgId => new ProviderOrganization { ProviderId = providerId, OrganizationId = orgId });
         var insertedProviderOrganizations = await _providerOrganizationRepository.CreateManyAsync(providerOrganizationsToInsert);
 
-        await _eventService.LogProviderOrganizationEventsAsync(insertedProviderOrganizations.Select(ipo => (ipo, EventType.ProviderOrganization_Added, (DateTime?)null)));
+        await _eventService.LogProviderOrganizationEventsAsync(
+            insertedProviderOrganizations.Select(ipo => (ipo, EventType.ProviderOrganization_Added, (DateTime?)null)));
+    }
+
+    private async Task UpdatePlanForExistingOrgAsync(IEnumerable<Guid> organizationIds, Provider provider)
```

Comment:
⛏️  Since this method looks for an existing provider instead of organization, a more descriptive name might be `UpdatePlanForExistingProviderAsync`.